### ISSUE TITLE
Safari extension loses keyboard focus

### ISSUE THREAD

**Issue by chmac** [bitwarden/clients#4765]
### Steps To Reproduce

This is a bit tricky to reproduce. It's based on timing...

1. Click on a link to a login page (the login page must have a `.focus()` call to put the keyboard focus into an input field)
2. Immediately open the bitwarden extension (should be already logged in but timed out so you see the password prompt)
3. Start typing your password

### Expected Result

If the input field of the webpage is activated via `.focus()`, the keyboard focus should stay in the password manager. While typing my password, the input focus should never change to another input field.

### Actual Result

If the web page calls `.focus()` on an input, my cursor will jump from the bitwarden extension password input to the web page's input field. If I'm not paying attention, I'll continue typing part of my password into an arbitrary web page.

### Additional Context

I realise that this might be an issue with Safari itself, and not specifically related to the bitwarden extension. I'm unsure if there is any way for the bitwarden extension to block the `.focus()` call from stealing the extension's focus. Maybe a workaround is possible by blocking access to input your password into bitwarden while the current tab is in a loading state or something similar?

If this issue isn't clear, just let me know and I'll try to expand. I can also create a test account and try to screen record an example of what I mean if necessary.

### Operating System

macOS

### Web Browser

Safari

### Build Version

Version: 2023.1.0

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by atjbramley** [comment]
Hi @chmac, 

Would you mind supplying an example website (or a selection) so that I can test that this occurs across all installations? 

Thanks!

**Comment by chmac** [comment]
@atjbramley  Sure thing. Here's an example site which has a slow delay (it's a single page app so takes a minute to load) and then fires a `.focus()` event https://www.libreview.com.

**Comment by chmac** [comment]
Hmmm, okay, it seems like I can't easily reproduce this consistently. I've tried 2 sites now, but they don't seem to produce this behaviour on demand. I've noticed this issue happen to me maybe only 3 times over the last month. I notice because all of a sudden my password is appearing in a plain text input field on my screen, so I've freaked out. But I can't figure out exactly what circumstances are required to make this happen. I'll keep an eye on this and try to find a more concrete reproduction.
